synapse/lib/view.py

```diff
@@ -300,6 +300,46 @@ def pack(self):
 
         return view
 
+    async def merge(self, user=None):
+        '''
+        Merge this view into its parent.  All changes made to this view will be applied to the parent.
+
+        When complete, delete this view.
+        '''
+        fromlayr = self.layers[0]
+        if self.parent is None:
+            raise s_exc.SynErr('Cannot merge a view than has not been forked')
+
+        if self.parent.layers[0].readonly:
+            raise s_exc.ReadOnlyLayer(mesg="May not merge if the parent's write layer is read-only")
+
+        for view in self.core.views.values():
+            if view.parent is not None and view.parent == self:
+                raise s_exc.SynErr(mesg='Cannot merge a view that has children itself')
+
+        CHUNKSIZE = 1000
+        fromoff = 0
+
+        if user is None:
+            user = self.core.auth.getUserByName('root')
+
+        await self.core.boss.promote('storm', user=user, info={'merging': self.iden})
```

Comment:
Is there a piece of book-keeping here that we need to do in order to check/kill any existing tasks which are using the view we're merging? or to lock out or set it to readonly somehow?

Reply:
We're deleting the view. I think there's an open issue re: what to do about tasks using an deleted view, but I think that's out of scope.